docs/configuration.md

```diff
@@ -0,0 +1,280 @@
+
+Each configuration section is as follows:
+* `instance_name` - A name for the configuration instance. Used for Samba's
+  server (netbios) name.
+* `instance_features` - Optional list. Feature names that alter the high
+  level behavior of sambacc. Valid feature flags are: `CTDB`, `ADDC`.
+* `shares` - List. The names of one or more share config sections to include
+  as part of the sambacc configuration.
+* `globals` - List. The names of one or more global config sections to
+  include as part of the sambacc configuration.
+* `domain_settings` - Only used with `ADDC` feature. Name of the AD DC
+  domain configuration.
+
+The subsections under `configs` can be used to uniquely identify one server
+"instance". Because those server instances may repeat the shares and samba
+globals are defined in their own sections and then included in an
+instance by referring to them in the `shares` and `globals` section
+of these subsections.
+
+
+## Shares Section
+
+The `shares` section is a mapping of share names to a share-configuration block.
+It is assumed that a configuration will have at least one share.
+
+Each share configuration section is as follows:
+* `options` - Mapping. The keys and values contained within are processed by
+  sambacc and become part of the smb.conf (or functional equivalent)
+  when running a Samba server.
+* `permissions` - Permissions configuration section:
+  * `method` - Permissions method. Known methods are:
+    * `none` - Perform no permissions management
+    * `initialize-share-perms` - Set share permissions only once. Track status in xattr.
+    * `always-share-perms` - Always set share permissions.
+  * `status_xattr` - Name of xattr to store status.
+  * Remaining key-value pairs are method specific. Unknown keys are ignored.
+  * `mode` - String that converts to octal. Unix permissions to set (`initialize-share-perms`, `always-share-perms`).
+
+
+## Globals Section
+
+The `globals` section is a mapping of named global configs to a
+globals-configuration block. It is assumed that a configuration will have
+at least one globals section.
+
+Each globals configuration section is as follows:
+* `options` - Mapping. The keys and values contained within are processed by
+  sambacc and become part of the global values in smb.conf (or functional
+  equivalent) when running a Samba server.
+
+If a configuration section names more than one globals section. All of the
+options within will be merged together to produce a single list of Samba
+configuration globals.
+
+
+## Users Section
+
+The `users` section defines local users for a non-domain-member server
+instance.
+
+The `users` section supports one key, `all_entries`, which is a list of
+user entries. Each user entry is as follows:
+* `name` - The user's name.
+* `password` - Optional. A plain-text password.
+* `nt_hash` - Optional. An NT-Hashed password.
+* `uid` - Optional integer. Specify the exact Unix UID the user should have.
```

Comment:
wondering if it would be helpful to state that the underlying unix uer will be created  with trhis ID. (at least I think it will) - just a thought.